src/main/webapp/src/app/location/location.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, Output, EventEmitter, ViewChild, ElementRef } from '@angular/core';
+import { LocationService } from '../location.service';
+import { Location } from '../location';
+import { string } from 'prop-types';
+
+@Component({
+  selector: 'app-location',
+  templateUrl: './location.component.html',
+  styleUrls: ['./location.component.css']
+})
+export class LocationComponent{
+
+  @Output() locationStored: EventEmitter<void> = new EventEmitter()
+	location = {} as Location;
+	locationFound = false;
+
+  @ViewChild("errorMessage", { static: false })
+  errorMessage!: ElementRef;
+
+  constructor(private locationService: LocationService) { }
+
+  handleLocationError(browserHasGeolocation: boolean, message: string): void {
+
+    browserHasGeolocation
+      ? this.errorMessage.nativeElement.innerText = "Geolocation services have failed. Try a default location. " + message
+      : this.errorMessage.nativeElement.innerText = "Geolocation is not supported by this browser. Try a default location."  
+
+  }
+
+  locationSuccess(position: Position): void {
+    this.location.lat = position.coords.latitude;
+    this.location.lng = position.coords.longitude;
+    this.setLocation();
+  }
+
+  locationError(error : PositionError): void {
```

Comment:
something weird seems to be going on with the formatting in this file, could you fix the indentation?